### PR TITLE
Implementing inverse operation

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

Implementing inverse of an NDArray. Issue [https://github.com/deepjavalibrary/djl/issues/1068](https://github.com/deepjavalibrary/djl/issues/1068)

### CODE REVIEW

File: engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtNDArray.java [hunk 1]
```diff
@@ -1386,6 +1386,11 @@ public PtNDArray erfinv() {
         return JniUtils.erfinv(this);
     }
 
+    @Override
```
Comment (frankfliu) [inline engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtNDArray.java:1389]:
```suggestion
    /** {@inheritDoc} */
    @Override
```

File: engines/tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java [hunk 2]
```diff
@@ -1337,6 +1337,11 @@ public NDArray isInfinite() {
         return manager.opExecutor("IsInf").addInput(this).buildSingletonOrThrow();
     }
 
+    @Override
```
Comment (frankfliu) [inline engines/tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java:1340]:
```suggestion
    /** {@inheritDoc} */
    @Override
```

File: api/src/main/java/ai/djl/ndarray/NDArray.java [hunk 3]
```diff
@@ -3644,6 +3644,13 @@ default NDArray argSort(int axis) {
      */
     NDArray isInfinite();
 
+    /**
+     * Inverses the {@code NDArray}.
+     *
+     * @return inverse {@code NDArray}.
+     */
```
Comment (siddvenk) [inline api/src/main/java/ai/djl/ndarray/NDArray.java:3651]: Should we specify that this will return the matrix inverse of a _square_ matrix if it exists?

Comment (sindhuvahinis) [reply api/src/main/java/ai/djl/ndarray/NDArray.java:3651]: Sure, Will specify it.

File: engines/mxnet/mxnet-engine/src/main/java/ai/djl/mxnet/engine/MxNDArray.java [hunk 4]
```diff
@@ -1225,6 +1225,12 @@ public NDArray isInfinite() {
         throw new UnsupportedOperationException("Not implemented yet.");
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public NDArray inverse() {
+        throw new UnsupportedOperationException("Not implemented yet.");
```
Comment (lanking520) [inline engines/mxnet/mxnet-engine/src/main/java/ai/djl/mxnet/engine/MxNDArray.java:1231]: https://mxnet.apache.org/versions/1.6/api/r/docs/api/mx.nd.reverse.html

Inverse NDArray

**Comment by sindhuvahinis** [comment]
@KexinFeng

**Review by frankfliu — APPROVED** [review]

**Pull request merged**